aqua/qt4-mac/Portfile

```diff
@@ -50,17 +50,28 @@ depends_lib-append  port:zlib \
 # value will be X.  The type is this variable is integer, so we can
 # use "==" and so forth for comparison.
 
-global MINOR
+global MINOR, MAJOR
+set MAJOR ""
 set MINOR ""
 
 # hopefully the MACOSX_DEPLOYMENT_TARGET exists and is set by now.  if
 # not, last resort (which is not desirable) is to use the os.version.
 
 if {${macosx_deployment_target} ne ""} {
+    set MAJOR [lindex [split ${macosx_deployment_target} "."] 0]
     set MINOR [lindex [split ${macosx_deployment_target} "."] 1]
 } else {
-    set MINOR [expr [lindex [split ${os.version} "."] 0] - 4]
+    # Do we really need this branch still ??
+    if { ${os.major} < 20 } {
+        set MAJOR 10
+        set MINOR [expr [lindex [split ${os.version} "."] 0] - 4]
+    } else {
+        # Bit of a guess, and will not work once we have macOS 12
+        set MAJOR 11
+        set MINOR [expr [lindex [split ${os.version} "."] 0] - 20]
+    }
 }
+ui_debug "Deduced OS MAJOR.MINOR = ${MAJOR}.${MINOR}"
```

Comment:
Is there a reason why the macOS version is being computed from the Darwin version here, rather than using the macOS version directly? We have `$macosx_version` (e.g. 10.13) for this now, and it will be replaced with `$macos_version` (e.g. 10.13.6 or 11.0.1) and `$macos_version_major`(e.g. 10.13 or 11) in MacPorts 2.7.0.

Reply:
I agree the version handling here should be reviewed, but my aim in this was to just do the minimal extension to what was there to allow builds on macOS 11. If the maintainer @michaelld wishes to make more significant changes here, I will leave that to him.